src/Microsoft.MobileBlazorBindings/MobileBlazorBindingsHostExtensions.cs

```diff
@@ -42,6 +44,40 @@ public static class MobileBlazorBindingsHostExtensions
             renderer.AddComponent<TComponent>(CreateHandler(parent, renderer)).ConfigureAwait(false);
         }
 
+        //Slight tweak on the above method to make it async and allow use with a serviceProvider instead of a host, and with a type argument instead of generic argument so it can be called at run time
+        //The above methos only needs host for it's services so it could be refactored.
+        //The above method only needs <t> so it can be used as a type in an overload of the method it calls.
+        //This version also allows an optional set of parameters
+        //The only downside is you can't have design/compiletime type safety
+        //There's a lot of duplicate code between the two, can probably refactor the core of the method into a separate method that they both call
+        public static async Task<IComponent> AddComponent(this IServiceProvider services, XF.Element parent, Type type, System.Collections.Generic.Dictionary<string,string> parameters = null)
+        {
+            if (services is null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+            if (parent is null)
+            {
+                throw new ArgumentNullException(nameof(parent));
+            }
+            if (type is null)
+            {
+                throw new ArgumentNullException(nameof(parent));
+            }
+            if(!typeof(IComponent).IsAssignableFrom(type))
+            {
+                throw new InvalidOperationException($"Cannot add {type.Name} to {parent.GetType().Name}. {type.Name} is not an IComponent. If you are trying to add an XF type, try adding the MobileBlazorBindings equivalent instead");
+            }
+
+            //NOTE: Do not add a using to dispose of this. If you do the page will look like it works but can't be updated after initial render.
```

Comment:
I think the comment is not needed because the warning is already disabled here.

Reply:
Fixed